### PR TITLE
Make sure we release adapters on failure

### PR DESCRIPTION
**Pull request by eriknordmark**
If a task fails to boot (hence it not activated) we still need to cleanup certain things on a delete.

### CODE REVIEW

**Review by mikem-zed — APPROVED** [review]
LGTM

File: pkg/pillar/cmd/domainmgr/domainmgr.go [hunk 1]
```diff
@@ -1767,7 +1770,7 @@ func doInactivate(ctx *domainContext, status *types.DomainStatus, impatient bool
 	status.IoAdapterList = nil
 	publishDomainStatus(ctx, status)
 
-	log.Functionf("doInactivate(%v) done for %s",
+	log.Functionf("doClanup(%v) done for %s",
```
Comment (zedi-pramodh) [inline pkg/pillar/cmd/domainmgr/domainmgr.go:1773]: Typo I guess

Comment (zedi-pramodh) [reply pkg/pillar/cmd/domainmgr/domainmgr.go:1773]: Otherwise LGTM

**Review by milan-zededa — APPROVED** [review]
Let's run tests

File: pkg/pillar/cmd/domainmgr/domainmgr.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/domainmgr/domainmgr.go]: Fixed

**Pull request merged**